### ISSUE TITLE
Add importlib.metadata as an approach in "Single-sourcing the package version"

### ISSUE THREAD

**Issue by chrahunt** [pypa/packaging.python.org#616]
The [current version](https://github.com/pypa/packaging.python.org/blob/50e680302eed35e2631d69f70881088e8d6124e6/source/guides/single-sourcing-package-version.rst) of "Single-sourcing the package version" has an example using `pkg_resources`. It would probably make sense to include an item using `importlib.metadata` since it will be standard in 3.8+ with a backport as `importlib_metadata` and can serve the same purpose. That may look like

```python
from importlib_metadata import version

__version__ = version(__package__)
```

**Comment by soxofaan** [comment]
I think this issue can be closed as it is already handled in 3c7c07f135303896a8c13a14697e136c99d4237b  (and followup tweaks)

**Comment by bhrutledge** [comment]
I think the documentation could be improved; looking at option 5 in https://packaging.python.org/en/latest/guides/single-sourcing-package-version/, I don't actually see a complete example of this technique.

**Comment by pradyunsg** [comment]
I suggest that we close this out and file a separate issue for improving the option 5 to de-emphasize `pkg_resources` and prefer `importlib.metadata` on 3.8+.

**Comment by ncoghlan** [comment]
Resolved in #1612
